datagateway_api/src/search_api/helpers.py

```diff
@@ -74,8 +84,76 @@ def create_error_message(e):
     return wrapper_error_handling
 
 
+def get_score(entities, query, group, limit):
+    """
+    Gets the score on the given entities based in the query parameter and limit
+    """
+
+    try:
+        myobj = {
+            "query": query,
+            "group": Config.config.search_api.scoring_group,
+            "limit": limit,
+            "itemIds": list(map(lambda entity: entity["pid"], entities)),
+        }
+        response = requests.post(
+            Config.config.search_api.scoring_server, json=myobj, timeout=5
+        )
+
+        if response.status_code < 400:
+            scores = response.json()["scores"]
+            print(str(len(scores)) + " scores retrieved from scoring app")
+            print(str(len(entities)) + " entities retrieved from ICAT")
+            for entity in entities:
+                items = list(
+                    filter(
+                        lambda score: str(score["itemId"]) == str(entity["pid"]),
+                        scores,
+                    ),
+                )
+
+                if len(items) == 1:
+
+                    print(
+                        "FOUND",
+                        entity["startDate"],
+                        entity["title"],
+                        entity["pid"],
+                        entity["doi"],
+                        entity["summary"],
+                        items[0]["score"],
+                    )
+                    entity["score"] = items[0]["score"]
+
+                else:
+                    # print("not FOUND %s", entity)
+                    print(
+                        "NOT FOUND",
+                        entity["startDate"],
+                        entity["title"],
+                        entity["pid"],
+                        entity["doi"],
+                    )
+                    entity["score"] = -1
+                # filter(
+                #    lambda x: x["itemId"] == entity["pid"], response.json()
+                # )["score"]
+            return entities
+        else:
+            raise ScoringAPIError(
+                Exception(f"Score API returned {response.status_code}"),
+            )
+    except ConnectionError as e:
+        log.error("ConnectionError to %s ", url)
+        raise e
+    except Exception as e:
```

Comment:
Are there any other specific exceptions that we could catch, instead of a generic `Exception`?